webapp/src/modules/parcels/utils.js

```diff
@@ -0,0 +1,9 @@
+import { isOnSale } from 'modules/asset/utils'
+import { hasParcelsConnected } from 'shared/parcel'
+
+export function canCreateEstate(wallet, parcel, publications) {
```

Comment:
Maybe we can use the `parcelsById` directly instead of passing the whole wallet

Reply:
True! but I think this might be a bit more flexible, as is a "high level function" receiving the top level resources ( `wallet, parcel, publications` ) and then delegating to more specific things:

```ts
!isOnSale(parcel, publications) &&
hasParcelsConnected(parcel, wallet.parcelsById)
```

This allows us to add more conditions later and still have the wallet available and also to be consistent in other parts of the code like this one:

```ts
switch (action) {
  case ACTIONS.bid:
    return isBiddeable(wallet, asset, bids)
  case ACTIONS.createEstate:
    return canCreateEstate(wallet, asset, publications)
  case ACTIONS.getMortgage:
    return canGetMortgage(wallet, asset, publications)
  default:
    return true
}
```

what do you think?

Reply:
Agree.